boudicca.events/eventcollectors/src/main/kotlin/events/boudicca/eventcollector/collectors/ValugCollector.kt

```diff
@@ -0,0 +1,75 @@
+package events.boudicca.eventcollector.collectors
+
+import base.boudicca.SemanticKeys
+import base.boudicca.api.eventcollector.Fetcher
+import base.boudicca.api.eventcollector.TwoStepEventCollector
+import base.boudicca.model.Event
+import base.boudicca.model.EventCategory
+import net.fortuna.ical4j.data.CalendarBuilder
+import net.fortuna.ical4j.model.Calendar
+import net.fortuna.ical4j.model.component.VEvent
+import java.io.StringReader
+import java.net.URI
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.*
+
+/**
+ * VorAlpen Linux User Group
+ */
+class ValugCollector : TwoStepEventCollector<VEvent>("valug") {
+    private val fetcher = Fetcher()
+    private val baseUrl = "https://valug.at/"
+    private val icsUrl = "${baseUrl}events/index.ics"
+
+    override fun getAllUnparsedEvents(): List<VEvent> {
+        val feed = fetcher.fetchUrl(icsUrl)
+        val builder = CalendarBuilder()
+        val calendar: Calendar = builder.build(StringReader(feed))
+        return calendar.components.filterIsInstance<VEvent>()
+    }
+
+    override fun parseEvent(event: VEvent): Event? {
+        val eventName = event.summary.value
+        val formatter = DateTimeFormatter.ofPattern("yyyyMMdd'T'HHmmss")
+        // get by city name to account for daylight saving time
+        val viennaZoneId = TimeZone.getTimeZone("Europe/Vienna").toZoneId()
+        val eventStartDate = LocalDateTime.parse(event.startDate.value, formatter)
+            .atZone(viennaZoneId).toOffsetDateTime()
+        val now = LocalDateTime.now().atZone(viennaZoneId).toOffsetDateTime()
+
+        if (eventStartDate.isBefore(now)) {
+            return null
+        }
+
+        val eventEndDate = LocalDateTime.parse(event.startDate.value, formatter)
+            .atZone(TimeZone.getTimeZone("Europe/Vienna").toZoneId()).toOffsetDateTime()
+
+        return Event(
+            eventName, eventStartDate,
+            buildMap {
+                put(SemanticKeys.ENDDATE, eventEndDate.format(DateTimeFormatter.ISO_DATE_TIME))
+                if (event.location != null) {
+                    put(SemanticKeys.LOCATION_NAME, event.location.value)
+                    if (event.location.value.contains("VALUG-Klubraum")) {
```

Comment:
adding additional location data should be done with our enricher, so that potentially other eventcollectors benefit from that information as well. how that is currently done is by entering this information into this google-sheet, if then the location name matches it will automatically add all the other information there as well: https://docs.google.com/spreadsheets/d/1yYOE5gRR6gjNBim7hwEe3__fXoRAMtREkYbs-lsn7uM/edit#gid=0
i have entered this information for you right now since we do not want to allow anyone to edit this data without moderation.
this means that you please should remove this if from the collector

Reply:
Great, thank you. I wondered how this would be done, but have not found it in the documentation.

Please add an alias for the location, since the location in the ical is mostly written as `VALUG-Klubraum, Alter Schl8hof Wels`.

Furthermore, the coordinates seem to be in the wrong order. The [type definition](https://github.com/boudicca-events/boudicca.events/blob/main/SEMANTIC_CONVENTIONS.md#data-types) states `<longitute>, <latitude>`, and [OSM](https://www.openstreetmap.org/api/0.6/node/1148138856) says the value starting with 14 is the longitude, but when I compare it to the rest of the entries (especially the other entry for schlachthof) it seems to be in the wrong order.

Reply:
i added the alias

and whoops, thanks for noticing!
i think i messed up the order in the docs, i think normally the default is latitude and then longitude, at least my googling shows me mainly this order :D